src/pages/UserSite/HomePage/homePage.tsx

```diff
@@ -23,19 +24,19 @@ export default class HomePage extends React.Component{
             allReviews: []
         }
 
-        this.handleSearchTerm = this.handleSearchTerm.bind(this);
+        this.handleSearchTerm = this.handleSearchTerm.bind(this); // note
         this.handleSubmit = this.handleSubmit.bind(this);
     }
 
     componentDidMount(){
-        const app: any = this;
+        const app = this; // note
         app.getData();
     }
 
     getData(){
         const app: any = this;
 
-        fetch(establishmentsApiUrl)
+        fetch(establishmentsApiUrl) // note
```

Comment:
Her hadde jeg laget en fetchJson funksjon som du kan bruke flere plasser. Da kan du også bruke generics sånn at du kan sette type på return verdien sånn at typescript kan gi beskjed hvis noe er feil.

```
function fetchJSON<T>(url: string): Promise<T> {
    return fetch(url)
        .then(r => r.json())
}
```

Så kan du calle den sånn f.eks: 
```
fetchJSON<Establisments[]>(establismentsApiUrl)
    .then(establisments => app.setState({establishments}))
```

Hvis da Establishments[] er feil type for state.establishments så klager typescript på deg.